convert2rhel/applock.py

```diff
@@ -103,8 +101,28 @@ def _try_create(self):
 
     @property
     def is_locked(self):
-        """Test whether this object is locked."""
-        return self._locked
+        """Test whether this object was locked by this instance of
+        the application."""
+        pid = self._read_pidfile()
+        if pid is not None:
+            return pid == self._pid
+        return False
+
+    def _read_pidfile(self):
+        """Read and return the contents of the PID file.
+
+        :returns: the file contents as an integer, or None if it doesn't exist
+        :raises: ApplicationLockedError if the contents are corrupt
+        """
+        if os.path.exists(self._pidfile):
```

Comment:
Note: To avoid races, it's better to do a `try: except:` around the `open()` rather than check for file existence first.